Makefile

```diff
@@ -85,7 +86,7 @@ $(OBJS_COMMON) $(OBJS_IR) $(OBJS_IR_TEST): $(BUILD_DIR)/$(notdir %.o): $(SRC_DIR
 	$(CC) $(CFLAGS) -I$(BUILD_DIR) -o $@ -c $<
 
 $(BUILD_DIR)/ir-test: $(SRC_DIR)/ir-test.cxx
-	$(CXX) -O3 -std=c++17 $(SRC_DIR)/ir-test.cxx -o $(BUILD_DIR)/ir-test
+	$(CXX12) -O3 -std=c++17 $(SRC_DIR)/ir-test.cxx -o $(BUILD_DIR)/ir-test
```

Comment:
I'm not sure. ``g++`` works fine for me, and I don't have ``g++-12`` installed.

Reply:
Passing like `make CXX=g++-12` seems fine enough. On some distro a symlink to `g++` might not be created automatically, or there could be multiple versions. Having the compiler version hardcoded like this seems not future proof.

Thanks

Reply:
Reverted commit 'Use g++-12 to build ir-test'

Reply:
Thanks for the further checks, @stkeke. I went double checking and seems indeed it's about the standard implementation integrity. Despite `C++17` is requested, seems the `multiline` flag is missing in some lower compiler version, even though it has some C++17 features.

It seems also, that some distro can backport features independently into an older compiler, so it might be hard to say if having a version `X` tells 100% some feature is absent.  Perhaps, it can be worked around with some other `std::regex` flags :/ For now, keeping it as simple as passing the `CXX=` arg seems appropriate.

Thanks!

Reply:
@weltling agree!